x-pack/filebeat/input/awscloudwatch/state_handler.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package awscloudwatch
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/elastic/beats/v7/filebeat/beater"
+	"github.com/elastic/beats/v7/libbeat/statestore"
+)
+
+const statePrefix = "filebeat::aws-cloudwatch::state::"
+
+type StorableState struct {
+	LastSyncEpoch int64 `json:"last_sync_epoch" struct:"last_sync_epoch"`
+}
+
+type stateHandler struct {
+	lock  sync.Mutex
+	store *statestore.Store
+}
+
+func createStateHandler(store beater.StateStore) (*stateHandler, error) {
+	st, err := store.Access("")
+	if err != nil {
+		return nil, fmt.Errorf("error accessing persistence store: %v", err)
+	}
+
+	return &stateHandler{store: st}, nil
+}
+
+func (s *stateHandler) GetState(forCfg config) (StorableState, error) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	var ss StorableState
+
+	id, err := generateID(forCfg)
+	if err != nil {
+		return ss, err
+	}
+
+	got, err := s.store.Has(id)
+	if err != nil {
+		return StorableState{}, err
+	}
+
+	if !got {
+		// Set to Epoch Zero, which is as if start from beginning
+		return StorableState{LastSyncEpoch: 0}, err
```

Comment:
nit: this is not an error path, would prefer to make the error return an explicit `nil` here

Reply:
Yup agree, fixes with commit https://github.com/elastic/beats/pull/43251/commits/6eefa925cc11224a61f1627bb72956fe97e510ab